libs/output-mapping/tests/DeferredTasks/Metadata/SchemaColumnMetadataTest.php

```diff
@@ -0,0 +1,129 @@
+use Keboola\OutputMapping\DeferredTasks\Metadata\SchemaColumnMetadata;
+use Keboola\OutputMapping\Mapping\MappingFromConfigurationSchemaColumn;
+use Keboola\StorageApi\Metadata;
+use Keboola\StorageApi\Options\Metadata\TableMetadataUpdateOptions;
+use PHPUnit\Framework\TestCase;
+
+class SchemaColumnMetadataTest extends TestCase
+{
+    private const TEST_TABLE_ID = 'in.c-testApply.table';
+    private const TEST_PROVIDER = 'keboola.sample-component';
+
+    public function applyDataProvider(): Generator
+    {
+        yield 'load at once' => [
+            'bulkSize' => 10,
+            'expectedApiCalls' => 1,
+            'expectedColumnsMetadata' => [
+                [
+                    'col1' => [
+                        [
+                            'columnName' => 'col1',
+                            'key' => 'key1',
+                            'value' => 'val1',
+                        ],
+                        [
+                            'columnName' => 'col1',
+                            'key' => 'key2',
+                            'value' => 'val2',
+                        ],
+                    ],
+                    'col2' => [
+                        [
+                            'columnName' => 'col2',
+                            'key' => 'KBC.description',
+                            'value' => 'col2 description',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        yield 'load in chunks' => [
+            'bulkSize' => 1,
+            'expectedApiCalls' => 2,
+            'expectedColumnsMetadata' => [
+                [
+                    'col1' => [
+                        [
+                            'columnName' => 'col1',
+                            'key' => 'key1',
+                            'value' => 'val1',
+                        ],
+                        [
+                            'columnName' => 'col1',
+                            'key' => 'key2',
+                            'value' => 'val2',
+                        ],
+                    ],
+                ],
+                [
+                    'col2' => [
+                        [
+                            'columnName' => 'col2',
+                            'key' => 'KBC.description',
+                            'value' => 'col2 description',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider applyDataProvider
+     */
+    public function testApply(int $bulkSize, int $expectedApiCalls, array $expectedColumnsMetadata): void
+    {
+        $metadataClientMock = $this->createMock(Metadata::class);
+        $metadataClientMock->expects(self::exactly($expectedApiCalls))
+            ->method('postTableMetadataWithColumns')
+            ->withConsecutive(...array_map(function (array $columnsMetadata): array {
```

Comment:
`withConsecutive` je deprecated a ve verzi 10 už nebude

V PS jsme teď striktní v tom, že když je něco deprecated, tak to do nových testů nepřidáváme, zbytečně bysme si pak komplikovali upgrade.

Bohužel PHPunit na to nemá nějakou pěknou náhradu. Tak to řešíme třeba takhle, přes `willReturnCallback`

https://github.com/keboola/platform-libraries/blob/86a5b820059766fb79b63485284cdfad3e8b651d/libs/input-mapping/tests/Helper/RealDevStorageTableRewriteHelperTest.php#L644